cmd/contour/serve.go

```diff
@@ -396,14 +396,7 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 			ResponseHeadersPolicy: &responseHeadersPolicy,
 		},
 	}
-	if clients.ResourcesExist(k8s.GatewayAPIResources()...) {
-		log.Debug("Gateway API found, adding Gateway API processor")
-		dagProcessors = append(dagProcessors, &dag.GatewayAPIProcessor{
-			FieldLogger: log.WithField("context", "GatewayAPIProcessor"),
-		})
-	} else {
-		log.Debug("Gateway API not found, not adding Gateway API processor")
-	}
```

Comment:
I think this section of code needs to stay where it is, since it needs to happen before adding the listener processor to the list (see comment just below) and then adding all the processors to the builder. I'd just add a check for the gateway being defined in addition to the gateway API resources existing.

Reply:
yup just found that and pushed a change to move that back to where it needs to be.